paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/service/QueryExecutorOperator.java

```diff
@@ -0,0 +1,126 @@
+ */
+
+package org.apache.paimon.flink.service;
+
+import org.apache.paimon.data.BinaryRow;
+import org.apache.paimon.data.BinaryString;
+import org.apache.paimon.data.GenericRow;
+import org.apache.paimon.data.InternalRow;
+import org.apache.paimon.disk.IOManager;
+import org.apache.paimon.io.DataFileMeta;
+import org.apache.paimon.io.DataFileMetaSerializer;
+import org.apache.paimon.service.network.NetworkUtils;
+import org.apache.paimon.service.network.stats.DisabledServiceRequestStats;
+import org.apache.paimon.service.server.KvQueryServer;
+import org.apache.paimon.table.FileStoreTable;
+import org.apache.paimon.table.Table;
+import org.apache.paimon.table.query.LocalTableQuery;
+import org.apache.paimon.types.DataTypes;
+import org.apache.paimon.types.RowType;
+
+import org.apache.flink.runtime.state.StateInitializationContext;
+import org.apache.flink.streaming.api.operators.AbstractStreamOperator;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+
+import java.net.InetSocketAddress;
+import java.util.Collections;
+import java.util.List;
+
+import static org.apache.paimon.utils.SerializationUtils.deserializeBinaryRow;
+
+/** Operator for query executor. */
+public class QueryExecutorOperator extends AbstractStreamOperator<InternalRow>
+        implements OneInputStreamOperator<InternalRow, InternalRow> {
+
+    private static final long serialVersionUID = 1L;
+
+    private final Table table;
+
+    private transient LocalTableQuery query;
+
+    private transient IOManager ioManager;
+
+    public QueryExecutorOperator(Table table) {
+        this.table = table;
+    }
+
+    public static RowType outputType() {
+        return RowType.of(DataTypes.INT(), DataTypes.INT(), DataTypes.STRING(), DataTypes.INT());
+    }
+
+    @Override
+    public void initializeState(StateInitializationContext context) throws Exception {
+        super.initializeState(context);
+        this.ioManager =
+                IOManager.create(
+                        getContainingTask()
+                                .getEnvironment()
+                                .getIOManager()
+                                .getSpillingDirectoriesPaths());
+        this.query = ((FileStoreTable) table).newLocalTableQuery().withIOManager(ioManager);
+        KvQueryServer server =
+                new KvQueryServer(
+                        getRuntimeContext().getIndexOfThisSubtask(),
+                        getRuntimeContext().getNumberOfParallelSubtasks(),
+                        NetworkUtils.findHostAddress(),
+                        Collections.singletonList(0).iterator(),
+                        1,
+                        1,
+                        query,
+                        new DisabledServiceRequestStats());
+
+        try {
+            server.start();
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Why catch? `initializeState` will throw Exception.

Reply:
It is Throwable, not Exception